src/sqlfluff/core/dialects/dialect_ansi.py

```diff
@@ -1509,6 +1516,37 @@ class SelectStatementSegment(BaseSegment):
 )
 
 
+@ansi_dialect.segment()
+class CTEDefinitionSegment(BaseSegment):
+    """A CTE Definition from a WITH statement.
+
+    `tab (col1,col2) AS (SELECT a,b FROM x)`
+    """
+
+    type = "common_table_expression"
+    match_grammar = Sequence(
+        Ref("SingleIdentifierGrammar"),
+        Bracketed(
+            Ref("SingleIdentifierListSegment"),
+            optional=True,
+        ),
+        "AS",
+        Bracketed(
+            # Checkpoint here to subdivide the query.
```

Comment:
Ephemeral instead of checkpoint right? (in the comment)

Reply:
Good catch